library/src/main/java/org/apache/apex/malhar/lib/spillable/SpillableComponentManager.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2016 Apache Software Foundation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.apex.malhar.lib.spillable;
+
+import org.apache.apex.malhar.lib.spillable.Spillable.SpillableComponent;
+
+import com.datatorrent.api.Component;
+import com.datatorrent.api.Context.OperatorContext;
+
+/**
+ * This is a factory for spillable data structures. This should be used as
+ * a component inside {@link com.datatorrent.api.Operator}s.
+ */
+public interface SpillableComponentManager extends Component<OperatorContext>, SpillableComponent
```

Comment:
Where is SplillableComponent?

Documentation is outdated. It is not the factory. Is this the Spillable Composite Component ?  If yes, then can we call it a spillable composite component (or something to that effect) and not a spillable component manger